### PR TITLE
Updated COCO tools. Fixed import of TracerStub during validation.

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

**Comment by druzhkov-paul** [comment]
@Ilya-Krylov What's the reason for making this change? Some linter issues?

File: tools/coco_limit_image_size.py [hunk 1]
```diff
@@ -56,12 +62,23 @@ def main():
             image = cv2.resize(image, new_size)
             resized += 1
 
+            for i in images_idx_to_annotations_idx[image_info['id']]:
+                content['annotations'][i]['bbox'] = [x / ratio for x in
+                                                     content['annotations'][i]['bbox']]
+                if content['annotations'][i]['segmentation']:
+                    content['annotations'][i]['segmentation'] = [
+                        [x / ratio for x in c] for c in content['annotations'][i]['segmentation']
+                    ]
```
Comment (druzhkov-paul) [inline tools/coco_limit_image_size.py]: Have you tried running this on MS COCO annotation files?

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by druzhkov-paul** [comment]
:mask:

File: tools/coco_limit_image_size.py (hunk 1 above)
Comment (Ilya-Krylov) [reply tools/coco_limit_image_size.py]: good catch, fixed

File: tools/coco_limit_image_size.py [hunk 2]
```diff
@@ -15,31 +15,60 @@
...
     args.add_argument('input_root', help='Path to input images root folder.')
+    args.add_argument('output_annotation', help='Path to output COCO annotation (*.json).')
     args.add_argument('output_root', help='Path to output images root folder.')
     args.add_argument('--limit_image_size', type=int, nargs=2, default=(1080, 1920),
                       help='Resize images with size greater than specified here size (h, w)! '
                            'keeping aspect ratio.')
     return args.parse_args()
 
 
+def resize_segmenation(segmentation, ratio, img_h, img_w):
```
Comment (druzhkov-paul) [inline tools/coco_limit_image_size.py]: `segmentation`

File: tools/coco_limit_image_size.py [hunk 3]
```diff
@@ -52,6 +81,17 @@ def main():
...
+                content['annotations'][i]['bbox'] = [x / ratio for x in
+                                                     content['annotations'][i]['bbox']]
+                if content['annotations'][i]['segmentation']:
+                    content['annotations'][i]['segmentation'] = resize_segmenation(
+                        content['annotations'][i]['segmentation'],
+                        ratio,
+                        image.shape[0],
+                        image.shape[1]
+                    )
             new_size = int(floor(image.shape[1] / ratio)), int(floor(image.shape[0] / ratio))
```
Comment (druzhkov-paul) [inline tools/coco_limit_image_size.py]: I guess `floor` is redundant here, but feel free to leave as is.

**Review by druzhkov-paul — APPROVED** [review]

File: tools/coco_limit_image_size.py (hunk 2 above)
Comment (Ilya-Krylov) [reply tools/coco_limit_image_size.py]: fixed

File: tools/coco_limit_image_size.py (hunk 3 above)
Comment (Ilya-Krylov) [reply tools/coco_limit_image_size.py]: fixed

**Comment by Ilya-Krylov** [comment]
run tests

**Pull request merged**